src/components/dashboard.vue

```diff
@@ -182,7 +181,7 @@
                 this.$data.incoming = request;
 
                 this.$data.incoming.action = "vote";
-                let entity_id = this.$data.incoming.id.split(".");
+                let entity_id = this.$data.incoming.params.id.split(".");
```

Comment:
incoming is an object, but initialized with empty string. Suggest to initialize with null.

Which class does `incoming` follow? Wondering if introduce proper encapsulation should be introduced

Reply:
good point on null, I'll rummage round and fix were appropriate.

incoming is an API request from the client injected with linking details (chain, origin, appName etc.). It can be anything really.

This is part of what I was telling you about needing to refactor more things to clean up.